app/scripts/modules/core/src/pipeline/config/pipelineConfigurer.js

```diff
@@ -363,18 +362,25 @@ angular
           .finally(() => this.setViewState({ loading: false }));
       };
 
+      this.applyUpdateTs = (toSave) => {
+        return $scope.application.pipelineConfigs.refresh(true).then((pipelines) => {
+          const latestFromServer = pipelines.find((p) => p.id === toSave.id);
+          if (latestFromServer && latestFromServer.updateTs) {
+            toSave.updateTs = latestFromServer.updateTs;
+            this.updatePipelineConfig({ updateTs: latestFromServer.updateTs });
+          }
+          setOriginal(toSave);
+          markDirty();
+        });
+      };
+
       this.savePipeline = () => {
         this.setViewState({ saving: true });
         const toSave = _.cloneDeep($scope.pipeline);
         PipelineConfigService.savePipeline(toSave)
-          .then(() => $scope.application.pipelineConfigs.refresh(true))
+          .then(() => this.applyUpdateTs(toSave))
```

Comment:
Is this racy?

It looks like `savePipeline` POSTs (but that results in an orchestration task), and we immediately get all pipelineConfigs?

I tried it and it seems fine, but I don't know if that is guaranteed.

Reply:
the [gate endpoint](https://github.com/spinnaker/gate/blob/master/gate-web/src/main/groovy/com/netflix/spinnaker/gate/controllers/PipelineController.groovy#L88) waits for front50 to return success or failure